src/aframe-components/viewer-mode.js

```diff
@@ -0,0 +1,299 @@
+/* global AFRAME, THREE, STREET */
+
+AFRAME.registerComponent('viewer-mode', {
+  schema: {
+    preset: {
+      type: 'string',
+      default: 'camera-path',
+      oneOf: ['locomotion', 'camera-path']
+    },
+    cameraPath: {
+      type: 'string',
+      default: 'circle',
+      oneOf: ['circle', 'forward', 'strafe'],
+      if: { preset: 'camera-path' }
+    },
+    cameraStartPosition: {
+      type: 'vec3',
+      default: { x: 0, y: 1.6, z: 0 }
+    }
+  },
+
+  init: function () {
+    // Store references to existing elements
+    this.cameraRig = document.querySelector('#cameraRig');
+    this.camera = document.querySelector('#camera');
+
+    if (!this.cameraRig) {
+      console.error(
+        'viewer-mode component: No camera rig found with id "cameraRig"'
+      );
+      return;
+    }
+
+    // Check if STREET.timer exists
+    if (typeof STREET === 'undefined' || !STREET.timer) {
+      console.warn(
+        'viewer-mode component: No STREET.timer found, camera will not move'
+      );
+    }
+
+    // Initialize basic camera path (just a simple circle for demo)
+    this.initBasicCameraPath();
+
+    // Flag to control tick execution for camera path mode
+    this.cameraPathActive = false;
+
+    // Set up the initial mode
+    this.setupMode(this.data.preset);
+  },
+
+  update: function (oldData) {
+    // If preset has changed, update the mode
+    if (oldData.preset !== this.data.preset) {
+      this.setupMode(this.data.preset);
+    }
+  },
+
+  setupMode: function (mode) {
+    // First disable all modes
+    this.disableAllModes();
+
+    // Then enable the selected mode
+    if (mode === 'locomotion') {
+      this.enableLocomotionMode();
+    } else if (mode === 'camera-path') {
+      this.enableCameraPathMode();
+    }
+    // Notify other components about the mode change
+    this.el.emit('viewer-mode-changed', { mode: mode });
+  },
+
+  disableAllModes: function () {
+    // Disable locomotion controls but preserve position
+    this.cameraRig.setAttribute('movement-controls', 'enabled: false');
+    this.cameraRig.setAttribute('cursor-teleport', 'enabled: false');
+    this.camera.setAttribute('look-controls', 'enabled: false');
+
+    // Disable camera path animation
+    this.cameraPathActive = false;
+  },
+
+  enableLocomotionMode: function () {
+    // Restore original locomotion controls
+    this.cameraRig.setAttribute('movement-controls', 'enabled: true');
+    this.cameraRig.setAttribute('cursor-teleport', 'enabled: true');
+    this.camera.setAttribute('look-controls', 'enabled: true');
+  },
+
+  enableCameraPathMode: function () {
+    // Enable camera path animation
+    this.cameraPathActive = true;
+
+    // Apply the correct initial position based on current time
+    // Only if STREET.timer is available
+    if (typeof STREET === 'undefined' || !STREET.timer) {
+      return; // Don't move if no timer
+    }
```

Comment:
That condition is hard to understand, do you want to check timerActive here?

Reply:
1eec09617176a312a0a1928fa0ea7bd0426224f7